src/main/java/de/tum/in/www1/artemis/security/annotations/EnforceEditor.java

```diff
@@ -0,0 +1,22 @@
+package de.tum.in.www1.artemis.security.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * This annotation is used to enforce that the user is an editor.
+ * It should only be used with endpoints starting with {@code /api/}
+ * <p>
+ * It's only addable to methods. The intention is that a developer can see the required role without the need to scroll up.
+ * This also prevents overrides of the annotation.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@PreAuthorize("hasRole('EDITOR')")
+public @interface EnforceEditor {
```

Comment:
```suggestion
public @interface EnforceAtLeastEditor {
```
I think the current naming scheme might be confusing to new developers. The name implies to me that the endpoint should be accessible by editors _exclusively_. That is not the case, though, I think. In our role hierarchy that means that editor or any role with higher permission-level (i.e. instructor, admin) can access it.

Same for the `Enforce{Instructor,Tutor,Student}` annotations.

---

For `EnforceAdmin` and `EnforceNothing` the current names make sense.

Reply:
It's not like this was different before, but I agree that it can be a nice improvement if we're already at it.